ddl/ddl_api.go

```diff
@@ -2997,11 +2997,20 @@ func needToOverwriteColCharset(options []*ast.TableOption) bool {
 // `ALTER TABLE ADD COLUMN (c1 INT, c2 INT)` is split into
 // `ALTER TABLE ADD COLUMN c1 INT, ADD COLUMN c2 INT`.
 func resolveAlterTableAddColumns(spec *ast.AlterTableSpec) []*ast.AlterTableSpec {
-	specs := make([]*ast.AlterTableSpec, len(spec.NewColumns))
-	for i, col := range spec.NewColumns {
+	specs := make([]*ast.AlterTableSpec, 0, len(spec.NewColumns)+len(spec.NewConstraints))
+	for _, col := range spec.NewColumns {
 		t := *spec
 		t.NewColumns = []*ast.ColumnDef{col}
-		specs[i] = &t
+		t.NewConstraints = []*ast.Constraint{}
+		specs = append(specs, &t)
+	}
+	for _, con := range spec.NewConstraints {
```

Comment:
Do we need to dd a comment here? Like "build add constraint spec".

Reply:
Updated.